File: easy/125_valid_palindrome/valid_palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        s = "".join(filter(str.isalnum, s)).lower()
        for i in range(len(s)//2):
            if s[i] != s[len(s)-i-1]:
                return False
        return True


if __name__ == "__main__":
    sol = Solution()
    print(sol.isPalindrome("abc.Ba"))
    print(sol.isPalindrome("abcde"))

File: leetcode/easy/125_valid_palindrome/valid_palindrome.cpp
#include <string>

class Solution {
public:
  bool isPalindrome(std::string s) {
    std::string processed_s = "";
    for (auto c : s) {
      if (isalnum(c)) {
        processed_s += tolower(c);
      }
    }
    if (processed_s.size() == 0) {
      return true;
    }
    auto start = 0;
    auto end = processed_s.size() - 1;
    while (start < end) {
      if (processed_s[start] != processed_s[end]) {
        return false;
      }
      start++;
      end--;
    }
    return true;
  }
};
